bin/isolated-functions.sh

```diff
@@ -12,12 +12,25 @@ fi
 # It _must_ preceed all the calls to die and assert.
 shopt -s expand_aliases
 
-assert() {
-	local x pipestatus=${PIPESTATUS[*]}
-	for x in ${pipestatus} ; do
-		[[ ${x} -eq 0 ]] || die "$@"
-	done
-}
+source "${PORTAGE_BIN_PATH}/eapi9-pipestatus.sh" || exit 1
+if ___eapi_has_pipestatus; then
+	pipestatus() {
+		__pipestatus "$@"
+	}
+fi
+
+if ___eapi_has_assert; then
+	assert() {
+		local x pipestatus=${PIPESTATUS[*]}
+		for x in ${pipestatus} ; do
+			[[ ${x} -eq 0 ]] || die "$@"
+		done
+	}
```

Comment:
I wonder if this should be simplified and use `__pipestatus` internally?
```suggestion
	assert() {
		__pipestatus || die "$@"
	}
```